target_snowflake/__init__.py

```diff
@@ -147,7 +147,9 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
             # Get schema for this record's stream
             stream = o['stream']
 
-            stream_utils.adjust_timestamps_in_record(o['record'], schemas[stream])
+            # Symon: adjust_timesptamps_in_record validates date/datetime/time fields and resets invalid values to max_date. 
+            # skipping this for performance optimization + our records are output of computation so dates are valid.
+            # stream_utils.adjust_timestamps_in_record(o['record'], schemas[stream])
```

Comment:
for small data using parquet extraction, records are already validated in tap-s3-parquet.
for large data using csv extraction, we should check spark date going out of python date range in spark step as it is common issue for other targets instead of validating in each target.